src/structural_elements/CCPACSFuselageWallSegment.cpp

```diff
@@ -0,0 +1,292 @@
+            }
+        }
+
+        gp_Pnt upper_new = base_point.Translated(u_pos*ext_vec);
+        gp_Pnt lower_new = base_point.Translated(u_neg*ext_vec);
+
+        if (count==nWallPositions-1) {
+            // flush last position to shape
+            if( p.GetShape() && GetFlushConnectionEnd().value_or(false) ) {
+                FlushPointAlongVec(p, x_vec, bboxSize, upper_new);
+                FlushPointAlongVec(p, x_vec, bboxSize, lower_new);
+            }
+        }
+
+        // create the wall segment from the four corner points lower, upper, lower_new, upper_new
+        if (count > 0 ) {
+            TopoDS_Face face = MakeFace(lower, lower_new, upper_new, upper);
+            builder_wall.Add(wall, face);
+
+            //first and last face are enlarged for the cutting tool
+            gp_Pnt lower_cut = lower;
+            gp_Pnt upper_cut = upper;
+            gp_Pnt lower_new_cut = lower_new;
+            gp_Pnt upper_new_cut = upper_new;
+            if (count==1) {
+                lower_cut = lower.Translated(-bboxSize*x_vec);
+                upper_cut = upper.Translated(-bboxSize*x_vec);
+            }
+            if (count==nWallPositions-1) {
+                lower_new_cut = lower_new.Translated(bboxSize*x_vec);
+                upper_new_cut = upper_new.Translated(bboxSize*x_vec);
+            }
+            face = MakeFace(lower_cut, lower_new_cut, upper_new_cut, upper_cut);
+            builder_cutter.Add(m_cutPlanes, face);
+        }
+
+        // store base_pnts and remember current lower and upper point for next position
+        base_pnts.push_back(base_point);
+        upper = upper_new;
+        lower = lower_new;
+        ++count;
+    }
+
+    // trim the wall
+    // Step 1/2: Trim the wall by the fuselage
+    {
+        TopoDS_Compound cut_wall;
+        builder_wall.MakeCompound(cut_wall);
+
+        TopoDS_Shape result = SplitShape(wall, fuselageShape);
+        TopTools_IndexedMapOfShape faceMap;
+        TopExp::MapShapes(result, TopAbs_FACE, faceMap);
+        for (int i = 0; i < faceMap.Extent(); ++i) {
+            // check if face center is on the interior of the fuselage
+            TopoDS_Face face = TopoDS::Face(faceMap(i+1));
+            gp_Pnt faceCenter = GetCentralFacePoint(face);
+
+            BRepClass3d_SolidClassifier clas3d(fuselageShape);
+            clas3d.Perform(faceCenter, Precision::Confusion());
+
+            if (clas3d.State() == TopAbs_IN) {
+                builder_wall.Add(cut_wall, face);
+            }
+        }
+        wall = cut_wall;
+    }
+
+    // Step 2/2: Cut the wall with bounding elements:
+    if (GetBoundingElementUIDs()) {
+        for (std::string bounding_element_uid : GetBoundingElementUIDs()->GetBoundingElementUIDs()) {
+            const CCPACSFuselageWallSegment& bounding_element = GetWalls().GetWallSegment(bounding_element_uid);
+            // TODO: Check: The order of whoch the walls are evaulated seem to matter here!
```

Comment:
Is this TODO still relevant?

Reply:
I don't think it is. This implementation should be the same as cutting all walls with all bounding elements first, which results in a lot of small subfaces. This set of subfaces is independent of the cutting order. Then we just collect those faces that contain a base point. I will remove the comment.

Reply:
done!